chromaui/test-archiver
# Add a file extension based on Content-Type header for files that have it

Issue: CAP-1204

## What Changed

This does three things:

1. Adds playwright tests for complete "E2E" testing of the test-archiver.
2. Refactors the mapping of source files to destination files into a class called the `SourceMapper`, which follows a builder pattern.
3. Fixes an issue where directories and files can have the same name by making files that are sent with a `Content-Type` header have a file extension.

## How to test

There are two ways:
1. Run `yarn test:playwright` and verify everything passes.
2. Reproduce the steps in https://linear.app/chromaui/issue/CAP-1204/eisdir-illegal-operation-on-a-directory and verify that they pass.

## Change Type
